libraries/rush-lib/src/api/RushConfiguration.ts

```diff
@@ -714,6 +783,16 @@ export class RushConfiguration {
       )
     );
 
+    this.tempSubspaceShrinkwrapFileName = (subspaceName: string): string => {
+      // TODO: do subspace name validation here
+      const fullSubspacePath: string = path.join(
+        this.commonTempSubspaceFolderRoot,
+        subspaceName,
+        this.shrinkwrapFilename
+      );
```

Comment:
`path.join` is actually pretty expensive. It's safe to just concatenate with slashes.

```suggestion
      const fullSubspacePath: string = `${this.commonTempSubspaceFolderRoot}/${subspaceName}/${this.shrinkwrapFilename}`;
```